### PR TITLE
Fix sync script to still pull in from a local registry.

### LINKED ISSUE #505: `make cluster-sync` doesn't use the code in the repository, but a quay.io image
After 263a6aca098f25dc1f9b0c5bdd656a91d951acb7
````
--- a/deploy/operator.yaml
+++ b/deploy/operator.yaml
@@ -39 +39 @@ spec:
-        image: quay.io/kubevirt/hyperconverged-cluster-operator:latest
+        image: quay.io/kubevirt/hyperconverged-cluster-operator:1.1.0
````

[This snippet](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/master/cluster/sync.sh#L15) responsible for using a local registry no longer works:
````
sed -i "s/image: quay\.io\/kubevirt\/hyperconverged-cluster-operator:latest/image: registry:5000\/kubevirt\/hyperconverged-cluster-operator:latest/g" _out/operator.yaml
````

There's a similar snippet in deploy.sh, too.

### PR DESCRIPTION
**Pull request by maya-r**
Recently we stopped using :latest in some YAML files, breaking
this functionality. Tolerate any tag.

Closes #505

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: cluster/sync.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ rm -rf _out/
 cp -r deploy _out/
 
 # Sed from docker.io to local registry
-sed -i "s/image: quay\.io\/kubevirt\/hyperconverged-cluster-operator:latest/image: registry:5000\/kubevirt\/hyperconverged-cluster-operator:latest/g" _out/operator.yaml
+sed -i "s/image: quay\.io\/kubevirt\/hyperconverged-cluster-operator.*/image: registry:5000\/kubevirt\/hyperconverged-cluster-operator:latest/g" _out/operator.yaml
```
Comment (tiraboschi) [inline cluster/sync.sh]: Can you please use `:.*$` ?

File: hack/deploy.sh [hunk 2]
```diff
@@ -47,7 +47,7 @@ if [ -n "${IMAGE_FORMAT}" ]; then
     HCO_IMAGE=`eval echo ${IMAGE_FORMAT}`
 fi
 
-sed -i "s#image: quay.io/kubevirt/hyperconverged-cluster-operator:latest#image: ${HCO_IMAGE}#g" _out/operator.yaml
+sed -i "s#image: quay.io/kubevirt/hyperconverged-cluster-operator.*#image: ${HCO_IMAGE}#g" _out/operator.yaml
```
Comment (tiraboschi) [inline hack/deploy.sh]: same

**Comment by maya-r** [comment]
/hold
(oops)

File: cluster/sync.sh [hunk 3]
```diff
@@ -12,7 +12,7 @@ rm -rf _out/
 cp -r deploy _out/
 
 # Sed from docker.io to local registry
```
Comment (orenc1) [inline cluster/sync.sh]: if you already here, can you please fix the comment?
```suggestion
# Sed from quay.io to local registry
```

File: hack/deploy.sh [hunk 4]
```diff
@@ -47,7 +47,7 @@ if [ -n "${IMAGE_FORMAT}" ]; then
     HCO_IMAGE=`eval echo ${IMAGE_FORMAT}`
 fi
 
-sed -i "s#image: quay.io/kubevirt/hyperconverged-cluster-operator:latest#image: ${HCO_IMAGE}#g" _out/operator.yaml
+sed -i "s|image: quay.io/kubevirt/hyperconverged-cluster-operator:.*$|image: ${HCO_IMAGE}|" _out/operator.yaml
```
Comment (orenc1) [inline hack/deploy.sh]: 'g' is missing at the end of the replacement clause.

**Comment by tiraboschi** [comment]
@maya-r can you please rebase this on latest master?

**Comment by tiraboschi** [comment]
Closed for https://github.com/kubevirt/hyperconverged-cluster-operator/pull/521

**Pull request closed without merging**